### PR TITLE
build: Update catalog API schema and client to 25e728ee

### PR DESCRIPTION
**Pull request by floxbot**

### CODE REVIEW

File: cli/flox-rust-sdk/src/models/lockfile.rs [hunk 1]
```diff
@@ -203,6 +203,7 @@ impl LockedPackageCatalog {
     ) -> Self {
         // unpack package to avoid missing new fields
         let catalog::PackageResolutionInfo {
+            catalog: _,
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/lockfile.rs:206]: nonblocking: should we put this in the lockfile? Is this going to be required at some point?

Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/lockfile.rs:206]: blocking: I think it's worth doing a `just gen-data -f` to validate this change (plus we said we'd do that whenever we did spec bumps)

**Review by mkenigs — APPROVED** [review]
Looks good after we regenerate mocks

File: cli/flox-rust-sdk/src/models/lockfile.rs (hunk 1 above)
Comment (billlevine) [reply cli/flox-rust-sdk/src/models/lockfile.rs:206]: Maybe, but I think there's a number of upcoming changes that will be needed.  I foresee a discussion and PR focused on just that in the very near future.

Comment (billlevine) [reply cli/flox-rust-sdk/src/models/lockfile.rs:206]: Done.  Thank you @mkenigs

**Pull request merged**